src/externalized/scripting/ScriptingExtensionsLua.cc

```diff
@@ -51,28 +50,49 @@ static void RegisterLogger();
 static void RegisterListener(std::string observable, std::string luaFunctionName);
 static void UnregisterListener(std::string observable, std::string key);
 
-void JA2Require(std::string scriptFileName)
+int StracciatellaLoadFileRequire(lua_State* L)
+{
+	ST::string path = sol::stack::get<std::string>(L);
+	STLOGD("Loading LUA script file: {}", path);
+
+	try {
+		AutoSGPFile file(GCM->openGameResForReading(SCRIPTS_DIR "/" + path));
+		std::string script = file->readStringToEnd().to_std_string();
+
+		luaL_loadbuffer(L, script.data(), script.size(), ST::format("@{}", path).c_str());
+		return 1;
+	} catch (const std::runtime_error& ex) {
+		auto errorMessage = ST::format("Error loading lua required file `{}`: {}", path, ex.what());
+		sol::stack::push(L, errorMessage.c_str());
+		return 1;
+	}
+}
+
+void RunEntryPoint()
 {
 	if (isLuaInitialized)
 	{
 		throw std::runtime_error("JA2Require is not allowed after initialization");
```

Comment:
This is probably useless now.

Reply:
BTW, my original thinking was to force the requiring, parsing and validation of all script files at init stage, so it fails early instead of bugging out in the middle of a game.

Reply:
Good point, we could still keep this, but I don't see the necessity to restrict things like that. I would rather add some documentation for conventions in Lua.

Another thing I noticed is that it probably makes sense to namespace scripts such as `/example-mod/data/scripts/example-mod/script.lua` (by convention). So that

- the chance that requires accidentially require things from other mods is minimized
- multiple mods can still require each others lua scripts, so a mod that provides basic lua functionality can be built upon (similarily to for example [HugsLib](https://steamcommunity.com/workshop/filedetails/?id=818773962) for Rimworld)

Reply:
The namespace convention is a good idea. Otherwise there will likely be filename clashes.

This should also apply to the entrypoint script, so each mod has a distinct entrypoint script like `$MODNAME/main.lua` or `$MODNAME.lua`.